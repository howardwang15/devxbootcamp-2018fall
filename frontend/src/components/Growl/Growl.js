import React from 'react';
import dateFormat from 'dateformat';

import './Growl.scss';

const Growl = ({id, userId, userName, text, createdAt}) => (
  <div className="growl-card">
    <div className="growl-meta">
      <p className="growl-user-name">{userName}</p>
      {createdAt && (
        <p className="growl-created-at">
          {dateFormat(createdAt, 'mmm d, yyyy h:mm tt')}
        </p>
      )}
    </div>
    <p className="growl-text">{text}</p>
  </div>
);

export default Growl;
