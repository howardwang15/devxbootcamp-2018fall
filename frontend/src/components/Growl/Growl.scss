@import 'style/color';

.growl-card {
  padding: 10px;

  .growl-text {
    font-size: 20px;
    color: $color-text-dark-1;
    line-height: 27.4px;
  }

  p {
    font-size: 16px;
    color: $color-text-dark-3;

    &.growl-user-name {
      font-weight: bold;
    }

    &.growl-text {
      color: $color-text-dark-1;
    }
  }

  .growl-meta {
    display: flex;
    & > * {
      margin-right: 4px;
    }
  }
}
