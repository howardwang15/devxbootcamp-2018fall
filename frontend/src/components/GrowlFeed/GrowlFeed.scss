.growl-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 3px;
  border: 1px solid rgba(0, 0, 0, 0.15);
  & > * {
    width: 100%;
    &:not(:last-child) {
      border-bottom: 1px solid rgba(0, 0, 0, 0.15);
    }
  }
}
