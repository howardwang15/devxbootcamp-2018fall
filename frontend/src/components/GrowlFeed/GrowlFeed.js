import React from 'react';
import {connect} from 'react-redux';

import {getGrowls} from 'actions';
import Growl from '../Growl';

import './GrowlFeed.scss';

export class GrowlFeed extends React.Component {
  static defaultProps = {
    growls: [],
    growlFilter: (growl) => true,
  };

  constructor(props) {
    super(props);
    // TODO: Implement
  }

  render() {
    // TODO: Implement using filter, map, and the Growl component imported above
    return null;
  }
}

export const mapStateToProps = (state) => ({
  // TODO: Implement
});

export default connect(
  mapStateToProps,
  {getGrowls},
)(GrowlFeed);
