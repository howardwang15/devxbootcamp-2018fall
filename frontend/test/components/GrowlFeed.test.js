import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Growl from '../../src/components/Growl/Growl';
import {GrowlFeed} from '../../src/components/GrowlFeed/GrowlFeed';

const MAKE_SAMPLE_GROWLS = () => [
  {
    id: '1',
    userId: '2',
    text: 'Test success',
    createdAt: new Date(
      'Tue Oct 09 2018 21:53:47 GMT-0700 (Pacific Daylight Time)',
    ),
  },
  {
    id: '2',
    userId: '3',
    text: 'Test fail',
    createdAt: new Date(
      'Tue Oct 09 2018 13:53:47 GMT-0700 (Pacific Daylight Time)',
    ),
  },
];

const props = {
  getGrowls: jest.fn(),
  growls: MAKE_SAMPLE_GROWLS(),
};

describe('GrowlFeed component', () => {
  it('renders matching the saved snapshot', () => {
    // Render to a tree that we can compare against our snapshot of the component
    const tree = renderer.create(<GrowlFeed {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('calls getGrowls when it initially renders', () => {
    // Render the component using Enzyme's shallow() tool
    const component = shallow(<GrowlFeed {...props} />);

    // Expect that GrowlFeed requested growls from Redux after it was constructed
    expect(props.getGrowls).toHaveBeenCalled();
  });

  it('renders correctly when filters growls as expected', () => {
    // Render the component using Enzyme's shallow() tool
    const component = shallow(
      <GrowlFeed {...props} growlFilter={(growl) => growl.id === '1'} />,
    );

    // Find all the Growl components
    const growlElements = component.find(Growl);
    // Ensure that it rendered only one Growl component for the one with id '1'
    expect(growlElements.length).toBe(1);
    expect(
      growlElements.everyWhere((growlEl) => growlEl.key() === '1'),
    ).toBeTruthy();
  });
});
