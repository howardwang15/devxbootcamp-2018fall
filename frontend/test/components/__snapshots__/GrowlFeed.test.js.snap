// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrowlFeed component renders matching the saved snapshot 1`] = `
<div
  className="growl-list"
>
  <div
    className="growl-card"
  >
    <div
      className="growl-meta"
    >
      <p
        className="growl-user-name"
      />
    </div>
    <p
      className="growl-text"
    >
      Test success
    </p>
  </div>
  <div
    className="growl-card"
  >
    <div
      className="growl-meta"
    >
      <p
        className="growl-user-name"
      />
    </div>
    <p
      className="growl-text"
    >
      Test fail
    </p>
  </div>
</div>
`;
